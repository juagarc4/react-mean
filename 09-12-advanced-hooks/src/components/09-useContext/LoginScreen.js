import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {
  const { setUser } = useContext(UserContext)
  const newUser = { id: 1, name: 'Raul', email: 'email@rgc.com' }
  return (
    <div>
      <h1>LoginScreen</h1>
      <hr />
      <button className="btn btn-primary" onClick={() => setUser(newUser)}>
        Login
      </button>
    </div>
  )
}
