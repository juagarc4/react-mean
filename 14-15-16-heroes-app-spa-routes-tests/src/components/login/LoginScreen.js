import React, { useContext } from 'react'
import { AuthContext } from 'auth/AuthContext'
import { types } from 'types/types'

export const LoginScreen = ({ history }) => {
  const { dispatch } = useContext(AuthContext)

  const handleLogin = () => {
    const lastVisited = localStorage.getItem('lastVisited') || '/'

    dispatch({
      type: types.login,
      payload: {
        name: 'Raul',
      },
    })
    history.replace(lastVisited)
  }
  return (
    <div className="container mt-5">
      <h1>Login Screen</h1>
      <hr />

      <button className="btn btn-primary" onClick={handleLogin}>
        Login
      </button>
    </div>
  )
}
