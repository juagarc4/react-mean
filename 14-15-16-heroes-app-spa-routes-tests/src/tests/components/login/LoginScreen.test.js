import { mount } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'
import { LoginScreen } from 'components/login/LoginScreen'
import { AuthContext } from 'auth/AuthContext'
import { types } from 'types/types'

describe('Tests in <LoginScreen/>', () => {
  const contextValue = {
    dispatch: jest.fn(),
    user: { logged: false },
  }

  const historyMock = {
    replace: jest.fn(),
  }

  const wrapper = mount(
    <AuthContext.Provider value={contextValue}>
      <MemoryRouter>
        <LoginScreen history={historyMock} />
      </MemoryRouter>
    </AuthContext.Provider>
  )
  afterEach(() => jest.clearAllMocks())

  test('Should render properly ', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('Should execute dispatch and navigation', () => {
    const handleClick = wrapper.find('button').prop('onClick')
    handleClick()
    const action = {
      type: types.login,
      payload: {
        name: 'Raul',
      },
    }

    expect(contextValue.dispatch).toHaveBeenCalledWith(action)
    expect(historyMock.replace).toHaveBeenCalledWith('/')

    localStorage.setItem('lastVisited', '/dc')
    handleClick()
    expect(historyMock.replace).toHaveBeenCalledWith('/dc')
  })
})
