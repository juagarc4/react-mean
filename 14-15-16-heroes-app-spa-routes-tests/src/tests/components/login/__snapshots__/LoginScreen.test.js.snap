// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <LoginScreen/> Should render properly  1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
